internal/engine/watchmanager.go

```diff
@@ -195,16 +195,27 @@ func (w *WatchManager) OnChange(ctx context.Context, st store.RStore) {
 			continue
 		}
 
+		filter, err := ignore.CreateFileChangeFilter(target)
+		if err != nil {
+			st.Dispatch(NewErrorAction(err))
+			return
+		}
+		tiltIgnoreFilter, err := dockerignore.DockerIgnoreTesterFromContents(tiltRoot, w.tiltIgnoreContents)
+		if err != nil {
+			st.Dispatch(NewErrorAction(err))
+		}
+		filter = model.NewCompositeMatcher([]model.PathMatcher{filter, tiltIgnoreFilter})
+
 		for _, d := range target.Dependencies() {
-			err = watcher.Add(d)
+			err = watcher.Add(d, filter)
 			if err != nil {
 				st.Dispatch(NewErrorAction(err))
 			}
 		}
 
 		ctx, cancel := context.WithCancel(ctx)
 
-		go w.dispatchFileChangesLoop(ctx, target, watcher, st, tiltRoot)
+		go w.dispatchFileChangesLoop(ctx, target, watcher, st, tiltRoot, filter)
```

Comment:
passing `filter` in here should be redundant now, since the watcher should now filter these events internally?

Reply:
It _should_ but the watchmanger tests depend on the watchmanager having this behavior in `dispatchFileChangesLoop` and trying to move that behavior in to the fake notifier is proving frustrating/flakey for reasons I'm not sure of.